Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -601,3 +601,23 @@ public enum ProductUpdateError: Error {
         }
     }
 }
+
+extension ProductUpdateError: Equatable {
+    public static func == (lhs: ProductUpdateError, rhs: ProductUpdateError) -> Bool {
+        switch (lhs, rhs) {
+        case (.duplicatedSKU, .duplicatedSKU):
+            return true
+        case (.invalidSKU, .invalidSKU):
+            return true
+        case (.notFoundInStorage, .notFoundInStorage):
+            return true
+        case (let .unknown(lhsError), let .unknown(rhsError)):
+            if let lhsDotcomError = lhsError as? DotcomError, let rhsDotcomError = rhsError as? DotcomError {
+                return lhsDotcomError == rhsDotcomError
+            }
+            return (lhsError as NSError) == (rhsError as NSError)
+        default:
+            return false
+        }
+    }
+}
```

Comment:
Suggestion: if we plan to reuse this pattern throughout the code maybe we could take advantage of some Type Erasures to not having to add manual equatable conformances to more enums.

If we define a struct like
```swift
public struct AnyError: Error {

    /// The underlying error.
    public let error: Error

    public init(_ error: Error) {
        if let anyError = error as? AnyError {
            self = anyError
        } else {
            self.error = error
        }
    }
}

extension AnyError: Equatable {
    public static func == (lhs: AnyError, rhs: AnyError) -> Bool {
        return (lhs as NSError) == (rhs as NSError)
    }
}
```

Then we can write `ProductUpdateError` or any other enum like

```swift
public enum ProductUpdateError: Error, Equatable {
    case duplicatedSKU
    case invalidSKU
    case notFoundInStorage
    case unknown(error: AnyError)
}
```

And have `equatable` conformance for free, what do you think?

Reply:
Love the `AnyError` implementation to get `Equatable` for free! I felt scared writing out the `Equatable` manually. Updated in https://github.com/woocommerce/woocommerce-ios/pull/2781/commits/7c49f0340f93e26c0ede8d41309f61d47213f0a5

Reply:
That's so cool!